StevesSixLeggedSprinters/BangazonAPI
# km-test1 Payment Type

Status

Choose:
READY

Migrations

YES

Description

I created a Payment model (PaymentType.cs) in order to create a table in the database for all payments. I also created a payment controller (PaymentTypeController.cs) so we are able to GET a list of payments, GET a single payment, POST (create) a payment, and PUT (edit) a specific payment. Also modified BangazonAPIContext file to set up a relationship between the database and the payment table.

Todos

[ Y] Tests
[ Y] Documentation
Deploy Notes

git fetch --a (fetch all branches from GitHub)
git branch -avv (display all branches)
git pull origin km-test1 (pull my code down to test)
dotnet ef migrations add Initial (migrations need to be built)
dotnet ef database update (update the database since a new migration has been built)
open bangazon.db (open database)
dotnet run

Steps to Test or Reproduce

I have set up the following methods: GET, POST, PUT, DELETE
Follow these step by step in order to test.

git fetch --a (fetch all branches from GitHub)
git branch -avv (display all branches)
git checkout km-test1 (switch into my branch)
dotnet ef migrations add Initial (migrations need to be built)
dotnet ef database update (update the database since a new migration has been built)
open bangazon.db (open database)
dotnet run

**Customer must be in the database to have Payment Entries, customer ID must match to add a payment type.**

Make sure to have Postman installed: https://www.getpostman.com/
In Postman,

To add a Payment Type:
Select POST
http://localhost:5000/api/PaymentType

And add this to the body before sending.
Example:
{
	"PaymentTypeId":"1",
	"AccountNumber":"5003235",
	"CustomerId":"1",
	"PaymentTypeName":"Bitcoin"
}


To view all departments:
Select GET
http://localhost:5000/api/PaymentType

To view a single Payment Type:
Select GET
http://localhost:5000/api/PaymentType/{id}

To edit a specific Payment Type
Select PUT
http://localhost:5000/api/PaymentType/{id}

To delete a Payment Type entry
Select Delete
http://localhost:5000/api/PaymentType/{id}

Impacted Areas in Application

List general components of the application that this PR will affect:
Orders will use the DeptTypeId as a foreign key.
*

https://github.com/StevesSixLeggedSprinters/BangazonAPI/issues/3
